com.unity.sg2/Editor/GraphUI/DataModel/ShaderGraphModel.cs

```diff
@@ -31,93 +32,40 @@ class MainPreviewData
         [NonSerialized]
         public float scale = 1f;
 
-        public void Initialize()
+        public Mesh mesh
         {
-            if (serializedMesh.IsNotInitialized)
-            {
-                // Initialize the sphere mesh as the default
-                Mesh sphereMesh = Resources.GetBuiltinResource(typeof(Mesh), $"Sphere.fbx") as Mesh;
-                serializedMesh.mesh = sphereMesh;
-            }
+            get => serializedMesh.mesh;
+            set => serializedMesh.mesh = value;
         }
     }
 
     public class ShaderGraphModel : GraphModel
     {
-        public GraphHandler GraphHandler => ShaderGraphAssetModel.GraphHandler;
-
-        public ShaderGraphAssetModel ShaderGraphAssetModel => Asset as ShaderGraphAssetModel;
-
-        public Registry RegistryInstance => ((ShaderGraphStencil)Stencil).GetRegistry();
-
-        #region MainPreviewData
-        MainPreviewData m_MainPreviewData = new ();
-
-        internal MainPreviewData mainPreviewData => m_MainPreviewData;
-
-        public void SetPreviewMesh(Mesh newPreviewMesh)
+        [SerializeField]
+        private SerializableGraphHandler graphHandlerBox = new();
+        [SerializeField]
+        private SerializableTargetSettings targetSettingsBox = new();
+        [SerializeField]
+        private MainPreviewData mainPreviewData = new();
+        [SerializeField]
+        private bool isSubGraph = false;
```

Comment:
The above is all the non-GTF serialization state that we care about. I think we still want to find a way to side-car it in the actual asset representation, but for undo/redo this is the cleanest thing we can do now.